### ISSUE TITLE
Missing secretsPathAmplifyAppId when trying Save and Deploy in DataStore

### ISSUE THREAD

**Issue by gmeligio** [aws-amplify/amplify-cli#8513]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-adminui/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-adminui/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d2tb4qmpt4etkw

### Region

eu-west-1

### Environment name

staging

### Amplify CLI Version

6.3.1

### If applicable, what version of Node.js are you using?

14.5.4

### What operating system are you using?

Windows

### Browser type?

Google Chrome

### Describe the bug

* The date/commit/version(s) of the code you're running
Date: 10/19/2021
Commit: 73fb4691b5d16de1cdcccf332c903f7f7d7af3ca

I enabled DataStore for the entire API in Amplify CLI and tried `Save and Deploy`, but that failed with the error `Parameters: [secretsPathAmplifyAppId] must have values`. Here is the log copied from Admin UI stack trace with sensitive data changed (the weird characters in Admin UI log should be because of incompatible text encoding between the terminal and the browser).

```
Command failed with exit code 1: /root/.amplify/bin/amplify push --yes --force
- Initializing SSM Client
- Fetching updates to backend environment: staging from the cloud.
✔ Successfully pulled backend environment staging from the cloud.
- Uploading files...
- Uploading files...
- Updating resources in the cloud. This may take a few minutes...
✖ An error occurred when pushing the resources to the cloud
⚠️ Amplify command hooks are disabled in the current execution environment.
⚠️ See https://docs.amplify.aws/cli/usage/command-hooks/ for more information.
[0m[0m
[0m    Current Environment: staging[0m
[0m    [0m
[90m┌──────────[39m[90m┬───────────────────────[39m[90m┬───────────[39m[90m┬───────────────────┐[39m
[90m│[39m Category [90m│[39m Resource name         [90m│[39m Operation [90m│[39m Provider plugin   [90m│[39m
[90m├──────────[39m[90m┼───────────────────────[39m[90m┼───────────[39m[90m┼───────────────────┤[39m
[90m│[39m Api      [90m│[39m myapp         [90m│[39m Update    [90m│[39m awscloudformation [90m│[39m
[90m├──────────[39m[90m┼───────────────────────[39m[90m┼───────────[39m[90m┼───────────────────┤[39m
[90m│[39m Auth     [90m│[39m MyApp         [90m│[39m No Change [90m│[39m awscloudformation [90m│[39m
[90m├──────────[39m[90m┼───────────────────────[39m[90m┼───────────[39m[90m┼───────────────────┤[39m
[90m│[39m Function [90m│[39m myappa8e2fe27 [90m│[39m No Change [90m│[39m awscloudformation [90m│[39m
[90m├──────────[39m[90m┼───────────────────────[39m[90m┼───────────[39m[90m┼───────────────────┤[39m
[90m│[39m Api      [90m│[39m api435a5c2e           [90m│[39m No Change [90m│[39m awscloudformation [90m│[39m
[90m└──────────[39m[90m┴───────────────────────[39m[90m┴───────────[39m[90m┴───────────────────┘[39m
[33m[39m
[33mThe following types do not have '@auth' enabled. Consider using @auth with @model[39m
[33m	 - Blog[39m
[33m	 - Post[39m
[33m	 - Comment[39m
[0mLearn more about @auth here: https://docs.amplify.aws/cli/graphql-transformer/auth[0m
[0m[0m
[32m[39m
[32mGraphQL schema compiled successfully.[39m

[32mEdit your schema at /tmp/amplify-a9752c10-189f-45af-ac6b-b822b8801e24/amplify/backend/api/myapp/schema.graphql or place .graphql files in a directory at /tmp/amplify-a9752c10-189f-45af-ac6b-b822b8801e24/amplify/backend/api/myapp/schema[39m


UPDATE_IN_PROGRESS amplify-amplifyb6087c3faed34-staging-212038 AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:42 GMT+0000 (Coordinated Universal Time) User Initiated
UPDATE_IN_PROGRESS functionmyappa8e2fe27               AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:47 GMT+0000 (Coordinated Universal Time)               


UPDATE_IN_PROGRESS          authMyApp                           AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:48 GMT+0000 (Coordinated Universal Time)                                                                                                
UPDATE_FAILED               functionmyappa8e2fe27               AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:48 GMT+0000 (Coordinated Universal Time) Parameters: [secretsPathAmplifyAppId] must have values                                         
UPDATE_FAILED               authMyApp                           AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:49 GMT+0000 (Coordinated Universal Time) Resource update cancelled                                                                      
UPDATE_ROLLBACK_IN_PROGRESS amplify-amplifyb6087c3faed34-staging-212038 AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:50 GMT+0000 (Coordinated Universal Time) The following resource(s) failed to update: [functionmyappa8e2fe27, authMyApp].


UPDATE_COMPLETE authMyApp             AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:56 GMT+0000 (Coordinated Universal Time) 
UPDATE_COMPLETE functionmyappa8e2fe27 AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:56 GMT+0000 (Coordinated Universal Time) 


UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS amplify-amplifyb6087c3faed34-staging-212038 AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:58 GMT+0000 (Coordinated Universal Time) 
UPDATE_COMPLETE                              authMyApp                           AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:59 GMT+0000 (Coordinated Universal Time) 
UPDATE_COMPLETE                              functionmyappa8e2fe27               AWS::CloudFormation::Stack Tue Oct 19 2021 09:55:59 GMT+0000 (Coordinated Universal Time) 
UPDATE_ROLLBACK_COMPLETE                     amplify-amplifyb6087c3faed34-staging-212038 AWS::CloudFormation::Stack Tue Oct 19 2021 09:56:00 GMT+0000 (Coordinated Universal Time) 


Following resources failed

🛑 
Resource is not in the state stackUpdateComplete
[31mAn error occurred during the push operation: Resource is not in the state stackUpdateComplete[39m
```

![image](https://user-images.githubusercontent.com/22875166/137889723-984c410c-e7c0-4f75-ab4b-fe6baf6fcce6.png)

### Expected behavior

That `Save and Deploy` completes successfully

### Reproduction steps

I'm not sure these steps can reproduce the problem but this is basically what I did in my project.

1. Have an existing Amplify App using the modules Auth, REST API and Function.
1. Create the GraphQL API `amplify api add`
1. Enable DataStore `amplify api update` and select `Enable DataStore for entire API`
1. Push changes `amplify push`
1. Go to Admin UI > Data.
2. Click on `Save and Deploy`

### Additional information

I checked in my project the file `amplify/team-provider-info.json`, and it has `"secretsPathAmplifyAppId": "d2tb4qmpt4etkw"` in the only Amplify Function that I have. Reviewing the bucket deployed to S3, it seems the `secretsPathAmplifyAppId` variable was also pushed when I ran `amplify push`.

**Comment by edwardfoyle** [comment]
@gmeligio Thanks for reporting this. This is happening because the `team-provider-info.json` file is not available in the Admin UI deployment job. While we work on a fix, you can work around the issue by deploying either using Amplify Console or using the Amplify CLI directly.

Alternatively, if you are only using one AWS account across all of your Amplify environments, you can move the `secretsPathAmplifyAppId` value from `team-provider-info.json` into `amplify/backend/function/<function name>/parameters.json`. The contents of this file are available during the Admin UI deployment and will be used to populate the CloudFormation parameters

**Comment by gmeligio** [comment]
I'm still having this issue. Can you help me?

Tried again "Save and Deploy" in Amplify Studio but it shows the same error.
<img width="1077" alt="image" src="https://user-images.githubusercontent.com/22875166/144666726-8c9f0d2b-29a2-4660-8867-614f72a568da.png">

**Comment by karan12990** [comment]
I am facing same error, I have auto deployment setup for my `feature/*` branches. I'm wondering how the secrets values will be available on auto deployment? Because I have added those values in cli.

**Comment by karan12990** [comment]
> @gmeligio Thanks for reporting this. This is happening because the `team-provider-info.json` file is not available in the Admin UI deployment job. While we work on a fix, you can work around the issue by deploying either using Amplify Console or using the Amplify CLI directly.
> 
> Alternatively, if you are only using one AWS account across all of your Amplify environments, you can move the `secretsPathAmplifyAppId` value from `team-provider-info.json` into `amplify/backend/function/<function name>/parameters.json`. The contents of this file are available during the Admin UI deployment and will be used to populate the CloudFormation parameters

@edwardfoyle is this fixed in latest version ?

**Comment by jaubrey-ebar** [comment]
**My scenario:** (in case this attracts other lost souls or helps fix the problem).
Everything was good in the amplify env/branch (ie dev) where I created the lambda function. Then I switched both my amplify env and git branch to beta, merged in the dev branch and pushed the merge. When the CI build picked up the change it failed to build. Digging in the logs showed "Parameters: [secretsPathAmplifyAppId] must have values". 

**My workaround:** 
1) run: `amplify update function` and then update any existing secret with an existing value.  This triggers amplify to resolve the config problem.  A git diff shows the following change in `amplify/team-provider-info.json`:

```
+        },
+        "MyFunctionName": {
+          "secretsPathAmplifyAppId": "MY_APP_ID",
+          "deploymentBucketName": "MY_BUCKET_NAME",
+          "s3Key": "amplify-builds/MY_BUILD_FILE_NAME-build.zip"
         }
```
2) Commit the changes and push. The CI build should now be successful.
